src/utils/router_state.rs

```diff
@@ -286,8 +298,10 @@ pub fn get_route(req: RouteQuery) -> Result<(Vec<Location>, f32), &'static str>
             ARROW_CARGO_ROUTER
                 .get()
                 .as_ref()
+                .ok_or("Can't access router")
                 .unwrap()
                 .get_node_by_id(*node_idx)
+                .ok_or(format!("Node not found by index {:?}", *node_idx))
```

Comment:
I thought `get_node_by_id` already handles error message? The `ok_or` is redundant. Is it possible to just use `.get_node_by_id(*node_idx)?.location`?

Reply:
Not possible because `get_node_by_id` returns Option, but get_route returns `Result`.
Using `?` is handy as it saves 4 lines of code using `match` (and avoids unwrap) but you can only use the operator if the function you call returns the same type as function where you call it. `ok_or` basically converts Option.Some to Result.OK, or None to Err.